app/hub/Views/HomeView/HomeViewContainer.jsx

```diff
@@ -61,75 +55,19 @@ class HomeViewContainer extends Component {
 		actions.setMetrics({ enable_metrics });
 	}
 
-	/**
-	 * Handle clicks on premium promo modal button
-	 * @param type		'basic' (default), 'plus', or 'premium'
-	 * @private
-	 */
-	_handlePremiumPromoModalClick = (type = 'basic') => {
-		// GH-1777
-		// we want to show the promo modal exactly once per Hub visit
-		const { actions } = this.props;
-		actions.markPremiumPromoModalShown();
-
-		sendMessage('SET_PREMIUM_PROMO_MODAL_SEEN', {});
-
-		switch (type) {
-			case 'plus':
-				window.open(`https://checkout.${DOMAIN}.com/plus?utm_source=gbe&utm_campaign=intro_hub`, '_blank');
-				break;
-			case 'premium':
-				window.open('https://ghostery.com/thanks-for-downloading-midnight?utm_source=gbe&utm_campaign=intro_hub', '_blank');
-				break;
-			case 'basic':
-			default:
-				break;
-		}
-	}
-
-	/**
-	 * @private
-	 * Function to handle clicks on "No thanks, continue with basic" in Premium promo modal
-	 */
-	_handleKeepBasicClick = () => { this._handlePremiumPromoModalClick(); }
-
-	/**
-	 * @private
-	 * Function to handle clicks on the "Get Plus instead" link in the Premium promo modal
-	 */
-	_handleGetPlusClick = () => { this._handlePremiumPromoModalClick('plus'); }
-
-	/**
-	 * @private
-	 * Function to handle clicks on the Midnight download button in the Premium promo modal
-	 */
-	_handleTryMidnightClick = () => { this._handlePremiumPromoModalClick('premium'); }
-
 	_render() {
 		const { justInstalled } = this.state;
-		const { home, user } = this.props;
-		const isPlus = (user && user.plusAccess) || false;
+		const { home, user, actions } = this.props;
+		const { sendPing } = actions;
 		const isPremium = (user && user.premiumAccess) || false;
 		const {
-			premium_promo_modal_shown,
 			setup_complete,
 			tutorial_complete,
 			enable_metrics,
 		} = home;
 
-		const showPromoModal = !premium_promo_modal_shown && !isPremium;
-
 		return (
 			<div className="full-height">
-				<PromoModal
-					type={PREMIUM}
-					show={showPromoModal}
-					isPlus={isPlus}
-					location="hub"
-					handleKeepBasicClick={this._handleKeepBasicClick}
-					handleGetPlusClick={this._handleGetPlusClick}
-					handleTryMidnightClick={this._handleTryMidnightClick}
-				/>
```

Comment:
Instead of removing this code, could you please restore it and put it behind a placeholder flag? The modal will in fact be displayed for one of the A/B/C test groups, so we still need the code in there. My A/B/C test logic will then set that flag and trigger display if needed based on which group the user belongs to

Reply:
Gotcha, that makes sense. I've added a flag and some comments